### PR TITLE
Feature/z user dropdown rework

### PR DESCRIPTION
**Pull request by DavideDA**
# Feature - z-user-dropdown - Rework for new topbar utilization

----

### Priority

- [ ] 1 - Highest
- [x] 2 - High
- [ ] 3 - Medium
- [ ] 4 - Low
- [ ] 5 - Lowest
- [ ] 6 - Not a Priority

### Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] Feature (non-breaking change which adds functionality)
- [ ] Component (add a Component as approved by Design System)
- [ ] Docs (add documentation)
- [ ] Chore (changes that adds small enhancement)
- [ ] Breaking (fix or feature that would cause existing functionality to not work as expected)

### Abstract

[https://app.abstract.com/projects/fd370780-e659-11e8-99dc-0d08537c5fde/branches/98773628-20c4-44d3-8184-ea44e47e0c57/commits/fe1969cd95b61dc23488d4d206b7e1866f4d6546/files/7e5534c1-96e2-44c2-98f8-965c34ee8835/layers/7BC1646B-F8EC-4466-A720-AEBAFFC3ABF0?collectionId=150f6337-8425-4fe7-9875-3fddc362f857&collectionLayerId=4fee133b-502a-4838-8b84-0fb438593cf9&mode=design](https://app.abstract.com/projects/fd370780-e659-11e8-99dc-0d08537c5fde/branches/98773628-20c4-44d3-8184-ea44e47e0c57/commits/fe1969cd95b61dc23488d4d206b7e1866f4d6546/files/7e5534c1-96e2-44c2-98f8-965c34ee8835/layers/7BC1646B-F8EC-4466-A720-AEBAFFC3ABF0?collectionId=150f6337-8425-4fe7-9875-3fddc362f857&collectionLayerId=4fee133b-502a-4838-8b84-0fb438593cf9&mode=design)

### Note

----

## Component and Fix approval flow to Master branch

A Pull Request to be merged must have two approvals, one from a member of the product team who developed it and another from a member of the dst dev team other than the team representative.

### CODE REVIEW

File: src/components/navigation/z-user-dropdown/index.tsx [hunk 1]
```diff
@@ -20,51 +12,53 @@ export class ZUserDropdown {
...
   }
 
   componentWillLoad() {
-    this.setMobileAndGhostDivWidth();
+    this.setMobileAndDivToResizeWidth();
   }
 
   componentDidLoad() {
-    this.setMobileAndGhostDivWidth();
+    this.setMobileAndDivToResizeWidth();
```
Comment (Abr89) [inline src/components/navigation/z-user-dropdown/index.tsx:39]: La componentDidLoad e la componentWillLoad fanno la stessa cosa, serve utilizzarle entrambe?
Avete provato a vedere se senza la willLoad continua a funzionare tutto?

File: src/components/navigation/z-user-dropdown/styles.css [hunk 2]
```diff
@@ -1,141 +1,269 @@
...
+  }
+
+  .userfullname.inverse {
+    color: var(--text-grey-800);
+  }
+
+  button.open div.userfullname {
+    color: var(--text-grey-800);
+    max-width: none;
+    text-overflow: clip;
```
Comment (Abr89) [inline src/components/navigation/z-user-dropdown/styles.css:249]: In caso di nomi molto lunghi il nome deve essere tagliato senza ellipsis?

File: src/components/navigation/z-user-dropdown/styles.css [hunk 3]
```diff
@@ -1,141 +1,269 @@
...
+  }
+
+  button.open div.userfullname.inverse {
+    color: var(--text-white);
+  }
+
+  .useremail {
+    margin-right: calc(var(--space-unit) * 3 + 2px);
```
Comment (Abr89) [inline src/components/navigation/z-user-dropdown/styles.css:257]: +2px?

File: src/components/navigation/z-user-dropdown/styles.css [hunk 4]
```diff
@@ -150,31 +278,31 @@ ul > li:last-child {
...
+    background-color: var(--bg-grey-800);
   }
 
-  ul > li > * {
-    width: max-content;
+  ul > li {
+    color: var(--text-grey-800);
+    border-bottom: var(--border-base) solid var(--gray400);
+    margin: 0 var(--space-unit);
+    padding-top: calc(var(--space-unit) * 2 - 2px);
```
Comment (Abr89) [inline src/components/navigation/z-user-dropdown/styles.css:290]: -2px?

File: src/components/navigation/z-user-dropdown/styles.css [hunk 5]
```diff
@@ -150,31 +278,31 @@ ul > li:last-child {
...
   }
 
+   ul > li.inverse {
+    color: var(--text-white);
+   }
+
   ul > li:first-child {
-    border-top: none;
+    border-top: var(--border-base) solid var(--gray400);
+    padding-top: calc(var(--space-unit) * 2 - 2px);
```
Comment (Abr89) [inline src/components/navigation/z-user-dropdown/styles.css:299]: (super minor) questo è superfluo perché come proprietà è già definita per il li

File: src/components/navigation/z-user-dropdown/index.tsx (hunk 1 above)
Comment (DavideDA) [reply src/components/navigation/z-user-dropdown/index.tsx:35]: Rimosso

File: src/components/navigation/z-user-dropdown/styles.css (hunk 2 above)
Comment (DavideDA) [reply src/components/navigation/z-user-dropdown/styles.css:249]: Non mi pare che tagli, sembra corretto il comportamento

**Review by Abr89 — APPROVED** [review]

**Pull request merged**